src/pageEditor/components/UrlPatternWidget.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2022 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from "react";
+import { useFormikContext } from "formik";
+import ArrayWidget from "@/components/fields/schemaFields/widgets/ArrayWidget";
+import FieldRuntimeContext from "@/components/fields/schemaFields/FieldRuntimeContext";
+import { PAGE_EDITOR_DEFAULT_BRICK_API_VERSION } from "@/pageEditor/extensionPoints/base";
+import { FormState } from "@/pageEditor/pageEditorTypes";
+import { SchemaFieldProps } from "@/components/fields/schemaFields/propTypes";
+import { Schema } from "@/core";
+
+/**
+ * https://developer.mozilla.org/en-US/docs/Web/API/URLPattern
+ *
+ * Leaving off the more obscure URL parts for now
+ */
+export const urlSchemaProject: Schema = {
+  type: "object",
+  properties: {
+    hostname: {
+      type: "string",
+      description:
+        "A string containing a pattern to match the hash part of a URL.",
```

Comment:
Typo
```suggestion
        "A string containing a pattern to match the hostname part of a URL.",
```

I don't know how the rest of the schema is described, but this feels bit verbose, especially in the UI. How about
```suggestion
        "Pattern to match the URL hostname.",
```